src/smart-components/Message/components/TextFragment/index.tsx

```diff
@@ -21,43 +22,37 @@ export default function TextFragment({
   const message = messageStore?.message as UserMessage;
   const isByMe = messageStore?.isByMe;
   const { updatedAt, createdAt } = message;
+
   return (
     <>
-      {
-        tokens?.map((token, idx) => {
-          const key = keyGenerator(createdAt, updatedAt, idx);
-          switch (token.type) {
-            case TOKEN_TYPES.mention:
-              return (
-                <span className="sendbird-word" key={key}>
-                  <MentionLabel
-                    mentionTemplate={USER_MENTION_PREFIX}
-                    mentionedUserId={token.userId}
-                    mentionedUserNickname={token.value}
-                    isByMe={isByMe}
-                  />
-                </span>
-              );
-            case TOKEN_TYPES.url:
-              return (
-                <span className='sendbird-word' key={key}>
-                  <LinkLabel
-                    className="sendbird-word__url"
-                    src={token.value}
-                    type={LabelTypography.BODY_1}
-                  >
-                    {token.value}
-                  </LinkLabel>
-                </span>
-              );
-            case TOKEN_TYPES.string:
-            default:
-              return (
-                <React.Fragment key={key}>{token.value}</React.Fragment>
-              );
-          }
-        })
-      }
+      {tokens?.map((token, idx) => {
+        const key = keyGenerator(createdAt, updatedAt, idx);
+        return (
+          <span className="sendbird-word" key={key}>
+            {match(token.type)
```

Comment:
Trying to bring ts-pattern \w a small example as we talked in this morning 
https://sendbird.slack.com/archives/C050Z6DA70A/p1681087655651419.